ts/src/binance.ts

```diff
@@ -2853,12 +2878,25 @@ export default class binance extends Exchange {
                 throw new ExchangeError (this.id + ' fetchMarkets() this.options fetchMarkets "' + marketType + '" is not a supported market type');
             }
         }
-        const promises = await Promise.all (promisesRaw);
+        const results = await Promise.all (promisesRaw);
         let markets = [];
-        for (let i = 0; i < fetchMarkets.length; i++) {
-            const promise = this.safeDict (promises, i);
-            const promiseMarkets = this.safeList2 (promise, 'symbols', 'optionSymbols', []);
-            markets = this.arrayConcat (markets, promiseMarkets);
+        this.options['crossMarginPairsData'] = {};
+        this.options['isolatedMarginPairsData'] = {};
+        for (let i = 0; i < results.length; i++) {
+            const res = this.safeValue (results, i);
+            if (fetchMargins && Array.isArray (res)) {
+                const keys = Object.keys (this.indexBy (res, 'symbol'));
+                const idsList = this.invertFlatStringDictionary (keys);
```

Comment:
supposedly this method expects a dict so not sure this is going to work in languages like c# when you do .Values over a list. WHy not use the list instead